lib/sqs-helper/index.js

```diff
@@ -0,0 +1,175 @@
+				...consumerProperties,
+				sqsName
+			}),
+
+			this.buildResource(sqsName, {
+				...mainQueueProperties,
+				dlqName
+			}),
+
+			this.buildResource(dlqName, dlqQueueProperties),
+
+			...Object.keys(dlqConsumerProperties).length ? [this.buildConsumerFunction(`${titleName}DLQ`, {
+				timeout: 15,
+				...dlqConsumerProperties,
+				sqsName: dlqName
+			})] : []
+
+		];
+	}
+
+	static validateConfigs(configs) {
+
+		if(!configs.name?.length)
+			throw new Error('Missing or empty name hook configuration in SQS helper');
+
+		[
+			['Main Consumer', configs.consumerProperties],
+			['Main Queue', configs.mainQueueProperties],
+			['DLQ Consumer', configs.dlqConsumerProperties],
+			['DLQ Queue', configs.dlqQueueProperties]
+		].forEach(([type, properties]) => {
+			if(properties && (typeof properties !== 'object' || Array.isArray(properties)))
+				throw new Error(`${type} Properties must be an Object with configuration in SQS helper`);
+		});
+	}
+
+	static getConfigsProperties(configs) {
+
+		const {
+			name,
+			dlqConsumerProperties = {}
+		} = configs;
+
+		const consumerProperties = {
+			...consumerDefaultsValue,
+			...configs.consumerProperties
+		};
+		const mainQueueProperties = {
+			...mainQueueDefaultsValue,
+			...configs.mainQueueProperties
+		};
+
+		const dlqQueueProperties = {
+			...dlqQueueDefaultsValue,
+			...configs.dlqQueueProperties
+		};
+
+		return {
+			name,
+			consumerProperties,
+			mainQueueProperties,
+			dlqQueueProperties,
+			dlqConsumerProperties
+		};
+	}
+
+	static buildConsumerFunction(functionName, {
+		timeout,
+		sqsName,
+		handler,
+		description,
+		functionProperties,
+		rawProperties,
+		...eventProperties
+	}) {
+
+		const fileName = `${functionName[0].toLowerCase()}${functionName.substring(1)}`;
+
+		return ['function', {
+			functionName: `${functionName}QueueConsumer`,
+			handler: handler || `src/sqs-consumer/${fileName}-consumer.handler`,
+			description: description || `${functionName} SQS Queue Consumer`,
+			timeout,
+			rawProperties: {
+				dependsOn: [sqsName],
+				...rawProperties
+			},
+			events: [
+				{
+					sqs: {
+						arn: `${baseArn}:\${self:custom.serviceName}${sqsName}`,
+						...eventProperties
+					}
+				}
+			],
+			...functionProperties
+		}];
+	}
+
+	static buildResource(name, {
+		maxReceiveCount,
+		dlqName,
+		receiveMessageWaitTimeSeconds,
+		visibilityTimeout,
+		messageRetentionPeriod,
+		...extraProperties
+	}) {
+
+		const queueName = `\${self:custom.serviceName}${name}`;
+
+		return ['resource', {
+			name,
+			resource: {
+				Type: 'AWS::SQS::Queue',
+				Properties: {
+					QueueName: queueName,
+					ReceiveMessageWaitTimeSeconds: receiveMessageWaitTimeSeconds,
+					VisibilityTimeout: visibilityTimeout,
+					// eslint-disable-next-line max-len
+					...dlqName && { RedrivePolicy: `{"maxReceiveCount": ${maxReceiveCount}, "deadLetterTargetArn": "${baseArn}:\${self:custom.serviceName}${dlqName}"}` },
```

Comment:
Quizás podriamos dejarlo más prolijo como ?
```suggestion
					...dlqName && {
						RedrivePolicy: JSON.stringify({
							maxReceiveCount,
							deadLetterTargetArn: `${baseArn}:\${self:custom.serviceName}${dlqName}`
						})
					},
```